server/lib/genome/groupers/drug_grouper.rb

```diff
@@ -4,53 +4,134 @@ class DrugGrouper < Genome::Groupers::Base
       attr_reader :term_to_match_dict
 
       def initialize
+        super
         url_base = ENV['THERAPY_HOSTNAME'] || 'http://localhost:8000'
-        if !url_base.ends_with? "/"
-          url_base += "/"
-        end
-        @normalizer_host = "#{url_base}therapy/"
-
-        @term_to_match_dict = {}
+        url_base += '/' unless url_base.ends_with? '/'
 
-        @sources = {}
+        @entity_type = 'drug'
+        @normalizer_entity_type = 'therapy'
+        @normalizer_host = "#{url_base}therapy/"
       end
 
-      def run(source_id = nil)
+      def group_claims(source_id = nil)
         claims = DrugClaim.eager_load(:drug_claim_aliases, :drug_claim_attributes).where(drug_id: nil)
         claims = claims.where(source_id: source_id) unless source_id.nil?
-        if source_id.nil?
-          puts "Grouping #{claims.length} ungrouped drug claims"
-        else
-          begin
-            source = Source.find(source_id)
-          rescue ActiveRecord::RecordNotFound
-            puts 'Unrecognized source ID provided'
-            return
-          end
-          source_name = source.source_db_name
-          puts "Grouping #{claims.length} ungrouped drug claims from #{source_name}"
-        end
 
+        print_grouping_message(claims, source_id)
         create_sources
 
-
         pbar = ProgressBar.create(title: 'Grouping drugs', total: claims.size, format: "%t: %p%% %a |%B|")
         claims.each do |drug_claim|
-          normalized_drug = normalize_claim(drug_claim.name, drug_claim.drug_claim_aliases)
-          next if normalized_drug.nil?
+          group_drug_claim(drug_claim)
+
+          pbar.progress += 1
+        end
+      end
+
+      private
+
+      def find_drug_attribute(drug_claim_attribute)
+        drug_attribute = DrugAttribute.where(
```

Comment:
I see that this was moved, not new code for this PR, so perhaps its not the right place to address it but aren't both branches of the `if` here checking the same thing, just expressed differently? Comparing `upper()` with `upcase` is effectively the same as comparing `lower()` with `downcase` unless there's some inconsistency with encoding somewhere.

Reply:
```ruby
      def find_drug_attribute(drug_claim_attribute)
        DrugAttribute.where('name ILIKE ? AND value ILIKE ?', drug_claim_attribute.name, drug_claim_attribute.value)
                     .first
      end
```

?